src/zoo.js

```diff
@@ -11,57 +11,128 @@ eslint no-unused-vars: [
 
 const data = require('./data');
 
-function animalsByIds(ids) {
-  // seu código aqui
+function animalsByIds(...ids) {
+  if (!ids) return [];
+  return ids.map(id => data.animals.find(animal => animal.id === id));
 }
 
-function animalsOlderThan(animal, age) {
-  // seu código aqui
-}
+const animalsOlderThan = (animal, age) =>
+  data.animals
+    .find(a => a.name === animal)
+    .residents.every(a => a.age > age);
 
-function employeeByName(employeeName) {
-  // seu código aqui
-}
+const employeeByName = employeeName =>
+  data.employees.find(
+    ({ firstName, lastName }) =>
+      firstName === employeeName || lastName === employeeName,
+  ) || {};
 
 function createEmployee(personalInfo, associatedWith) {
-  // seu código aqui
+  return { ...personalInfo, ...associatedWith };
 }
 
 function isManager(id) {
-  // seu código aqui
+  return data.employees.some(({ managers }) => managers.includes(id));
 }
 
-function addEmployee(id, firstName, lastName, managers, responsibleFor) {
-  // seu código aqui
+function addEmployee(
+  id,
+  firstName,
+  lastName,
+  managers = [],
+  responsibleFor = [],
+) {
+  const newEmployee = { id, firstName, lastName, managers, responsibleFor };
+  data.employees.push(newEmployee);
 }
 
 function animalCount(species) {
-  // seu código aqui
+  const allAnimals = data.animals.reduce((acumulador, atual) => {
+    acumulador[atual.name] = atual.residents.length;
+    return acumulador;
+  }, {});
+  if (!species) return allAnimals;
+  return data.animals.find(animal => animal.name === species).residents
+    .length;
 }
 
 function entryCalculator(entrants) {
-  // seu código aqui
+  return !entrants || Object.keys(entrants).length === 0
+    ? 0
+    : Object.keys(entrants).reduce(
+        (acc, curr) => acc + (entrants[curr] * data.prices[curr]),
+        0,
+      );
```

Comment:
Só mostrando uma outra forma de fazer, sem o .reduce e usando so object destructuring: 
```suggestion
  if (!entrants || Object.keys(entrants).length === 0) return 0;
  const { Adult: adultPrice, Child: childPrice, Senior: seniorPrice } = data.prices;
  const { Adult, Child, Senior } = entrants;
  return (adultPrice * Adult) + (childPrice * Child) + (seniorPrice * Senior);
```

Reply:
Eu tinha feito bem parecido... mas não estava dando certo. Obrigado pelo feedback... agora eu sei o que estava errado!